climate-monitoring-client/src/main/java/climatemonitoring/Registration.java

```diff
@@ -1,18 +1,154 @@
 package climatemonitoring;
 
+import climatemonitoring.core.ConnectionLostException;
+import climatemonitoring.core.DatabaseRequestException;
+import climatemonitoring.core.Operator;
 import climatemonitoring.core.ViewState;
+import climatemonitoring.core.headless.Console;
+import climatemonitoring.core.utility.Email;
+
+import java.util.Random;
 
 class Registration extends ViewState {
 
 	@Override
 	public void onHeadlessRender(String args) {
 
-		throw new UnsupportedOperationException("Unimplemented method 'onHeadlessRender'");
+		String errorMsg = null;
+
+		do {
+
+			m_userID = Console.read("User ID > ");
+			errorMsg = Check.userID(m_userID);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_email = Console.read("Email > ");
+			errorMsg = Check.emailUnique(m_email);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_password = Console.read("Password > ");
+			errorMsg = Check.password(m_password);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_SSID = Console.read("SSID > ");
+			errorMsg = Check.ssidUnique(m_SSID);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_name = Console.read("Name > ");
+			errorMsg = Check.name(m_name);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_surname = Console.read("Surname > ");
+			errorMsg = Check.surname(m_surname);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_centerID = Console.read("Center ID > ");
+			errorMsg = Check.centerID(m_centerID);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		try {
+
+			CenterCreation cc = null;
+
+			if (m_centerID == null) {
+
+				cc = (CenterCreation) Handler.getView().getState(ViewType.CENTER_CREATION);
+				cc.onHeadlessRender("");
+				m_centerID = cc.newCenter.getCenterID();
+			}
+	
+			Operator operator = new Operator(m_userID, m_SSID.toCharArray(), m_surname, m_name, m_email, m_password, m_centerID);
+
+			// Verification
+			Email email = new Email("climatemonitoringappservice@mail.com", "Climate Monitoring");
+			email.setReceiverEmail(m_email);
+			email.setReceiverName(m_name + " " + m_surname);
+			email.setSubject("Climate Monitoring verification code");
+
+			Random r = new Random();
+			int codeGiven = r.nextInt();
+			email.setMessage("Your verification code is: " + codeGiven + "\nIt will expire in 2 minutes");
+
+			long start = System.nanoTime();
+			int codeReceived = Integer.parseInt(Console.read("Your verification code > "));
+			long end = System.nanoTime();
```

Comment:
You forgot to actually send the email. You should write at line 111 something like this:
```
Result<Boolean> result = email.send();
result.join();
```

Reply:
Fixed